packages/web/src/components/social-proof/SocialProof.tsx

```diff
@@ -0,0 +1,227 @@
+  const record = useRecord()
+  const onTwitterClick = useCallback(() => {
+    onClick()
+    const trackEvent: TrackEvent = make(Name.SOCIAL_PROOF_OPEN, {
+      handle,
+      type: 'twitter'
+    })
+    record(trackEvent)
+  }, [record, handle, onClick])
+
+  const onInstagramClick = useCallback(() => {
+    onClick()
+    const trackEvent: TrackEvent = make(Name.SOCIAL_PROOF_OPEN, {
+      handle,
+      type: 'instagram'
+    })
+    record(trackEvent)
+  }, [record, handle, onClick])
+
+  return (
+    <div className={styles.container}>
+      <p>{messages.description}</p>
+      <div className={styles.btnContainer}>
+        <TwitterAccountVerification
+          onSuccess={onTwitterLogin}
+          onFailure={(error: Error) => onFailure('twitter', error)}
+          className={styles.twitterClassName}
+          onClick={onTwitterClick}
+        />
+        {displayInstagram && (
+          <InstagramAccountVerification
+            onClick={onInstagramClick}
+            onSuccess={onInstagramLogin}
+            onFailure={(error: Error) => onFailure('instagram', error)}
+          />
+        )}
+      </div>
+      {error && (
+        <div className={styles.error}>
+          <IconValidationX className={styles.validationIcon} />
+          {error}
+        </div>
+      )}
+    </div>
+  )
+}
+
+const LoadingBody = () => {
+  return (
+    <div className={styles.container}>
+      <LoadingSpinner className={styles.loadingContainer} />
+    </div>
+  )
+}
+
+type SocialProofProps = {
+  onSuccess: () => void
+}
+
+/**
+ * A modal that verifies social proof of a user.
+ * When proof is made, the modal closes. There are no
+ * callbacks so that a server may determine proof.
+ */
+const SocialProof = ({ onSuccess }: SocialProofProps) => {
+  const dispatch = useDispatch()
+  const onInstagramLogin = useCallback(
+    (uuid: string, profile: InstagramProfile) => {
+      dispatch(instagramLoginAction({ uuid, profile }))
+    },
+    [dispatch]
+  )
+  const onTwitterLogin = useCallback(
+    (uuid: string, profile: TwitterProfile) => {
+      dispatch(twitterLoginAction({ uuid, profile }))
+    },
+    [dispatch]
+  )
+
+  const [error, setError] = useState('')
+  const [status, setStatus] = useState('')
+  const [isOpen, setIsOpen] = useModalState('SocialProof')
+  const handle = useSelector(getUserHandle)
+
+  const onClick = useCallback(() => setStatus(Status.LOADING), [setStatus])
+  const record = useRecord()
+  const onFailure = useCallback(
+    (type: 'instagram' | 'twitter', error: Error) => {
+      if (!handle) return
+
+      setError(messages.failure)
+      setStatus(Status.ERROR)
+
+      const trackEvent: TrackEvent = make(Name.SOCIAL_PROOF_ERROR, {
+        handle,
+        type,
+        error: error.message
+      })
+      record(trackEvent)
+    },
+    [record, setError, setStatus, handle]
+  )
+
+  const instagramLogin = useCallback(
+    (uuid: string, profile: InstagramProfile) => {
+      if (!handle) return
```

Comment:
not sure of full context here, but do we want this to trigger failure notice if we dont have handle?